sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -1078,26 +1080,50 @@ object JdbcUtils extends Logging with SQLConfHelper {
    */
   def checkIfIndexExists(
       conn: Connection,
-      indexName: String,
       sql: String,
-      indexColumnName: String,
       options: JDBCOptions): Boolean = {
     val statement = conn.createStatement
     try {
       statement.setQueryTimeout(options.queryTimeout)
       val rs = statement.executeQuery(sql)
-      while (rs.next()) {
-        val retrievedIndexName = rs.getString(indexColumnName)
-        if (conf.resolver(retrievedIndexName, indexName)) {
-          return true
-        }
-      }
-      false
+      rs.next
+    } catch {
+      case _: Exception =>
+        logWarning("Cannot retrieved index info.")
+        false
     } finally {
       statement.close()
     }
   }
 
+  /**
+   * Process index properties and return tuple of indexType and list of the other index properties.
+   */
+  def processIndexProperties(
+      properties: util.Map[String, String],
+      options: JDBCOptions
+    ): (String, Array[String]) = {
+    val dialect = JdbcDialects.get(options.url)
+    var indexType = ""
+    var indexPropertyList: Array[String] = Array.empty
+
+    if (!properties.isEmpty) {
+      properties.asScala.foreach { case (k, v) =>
+        if (k.equals(SupportsIndex.PROP_TYPE)) {
+          if (v.equalsIgnoreCase("BTREE") || v.equalsIgnoreCase("HASH")) {
+            indexType = s"USING $v"
+          } else {
+            throw new UnsupportedOperationException(s"Index Type $v is not supported." +
+              " The supported Index Types are: BTREE and HASH")
```

Comment:
Seems PostgreSQL can support more index types (https://www.postgresql.org/docs/9.5/sql-createindex.html).

Reply:
@huaxingao, `GIST` and `GIN` index types in PostgreSQL are used for specific types: `tsvector` and `tsquery`, which are not supported types in Spark. Also for `SP-GIST` are used for the `point` type. Should we ignore these index types or something else? [GIST and GIN index type](https://www.postgresql.org/docs/9.1/textsearch-indexes.html) for details.